service/storage_service.cc

```diff
@@ -620,7 +620,10 @@ void storage_service::join_token_ring(int delay) {
         // (post CASSANDRA-1391 we don't expect this to be necessary very often, but it doesn't hurt to be careful)
         while (!get_local_migration_manager().have_schema_agreement()) {
             set_mode(mode::JOINING, "waiting for schema information to complete", true);
-            sleep(std::chrono::seconds(1)).get();
+            auto nodes = gossiper.get_live_members();
+            auto live_nodes = std::vector<gms::inet_address>(nodes.begin(), nodes.end());
+            service::get_local_migration_manager().sync_schema(_db.local(), live_nodes).get();
```

Comment:
If we're using `sync_schema()`, we can get rid of the loop and just sync with all the nodes once. The loop is needed if you're waiting for the background sync to do its job.

Reply:
If the first call to sync_schema actually makes the schema sync, we will call sync_schema once anyway. If not, to be safe, I do not see any problem to call it again.

Reply:
There is no problem with calling it in a loop, I just think it's not necessary. And the simpler the code the better.